### PR TITLE
[JIT] add unsupported section

### PR DESCRIPTION
**Pull request by eellison**
Add a section for unsupported ops, and modules. Automatically generate the properties and attributes that aren't bound, and for ops that have semantic mismatches set up tests so the docs stay up to date.

### CODE REVIEW

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 72bd412:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=72bd412f649f35aa6d55093d5333ff30812487f5).

### 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/3967287) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75290604) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75290604))
<details>
<summary>
<code>Automatic merge failed; fix conflicts and then commit the result.</code>
</summary>

```
CONFLICT (add/add): Merge conflict in .circleci/validate-docker-version.py 
Auto-merging .circleci/validate-docker-version.py 
CONFLICT (add/add): Merge conflict in .circleci/scripts/should_run_job.sh 
Auto-merging .circleci/scripts/should_run_job.sh 
CONFLICT (add/add): Merge conflict in .circleci/generate_config_yml.py 
Auto-merging .circleci/generate_config_yml.py 
CONFLICT (add/add): Merge conflict in .circleci/docker/common/install_travis_python.sh 
Auto-merging .circleci/docker/common/install_travis_python.sh 
CONFLICT (add/add): Merge conflict in .circleci/config.yml 
Auto-merging .circleci/config.yml 
Automatic merge failed; fix conflicts and then commit the result. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: docs/source/jit_unsupported.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+TorchScript Unsupported Pytorch Constructs
+============================================
+
+Torch and Tensor Unsupported Attributes
+------------------------------------------
+
+
+TorchScript supports most methods defined on `torch` and `torch.Tensor`, but we do not have full coverage.
+Here are specific known ops and categories of ops which have diverging behavior between
+Python and TorchScript. Deprecated ops are not listed below.
```
Comment (driazati) [inline docs/source/jit_unsupported.rst]: Why not list deprecated ops? We are detecting them below, we could also tag them with something  like `* resize (this method is deprecated and won't be supported)`

**Review by driazati — APPROVED** [review]
Seems like a good start but it's missing a lot of stuff from the quip (which itself is missing some other stuff like the unsupported parts of `torch.nn.init`)

File: torch/jit/unsupported_tensor_ops.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+def _gen_unsupported_methods_properties():
+    tensor_attrs = set(filter(lambda x: x[0] != "_", dir(torch.Tensor)))
+    tensor = torch.tensor([2])
+    funcs_template = dedent('''
+    def func(x):
+        return x.{op}()
+    ''')
+
+    deprecated_apis = set(["volatile", "resize", "reinforce", "new", "name", "map2_", "has_names", "grad_fn", "resize_as"])
```
Comment (driazati) [inline torch/jit/unsupported_tensor_ops.py]: Also where is this list coming from?

File: docs/source/jit_unsupported.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+PyTorch Unsupported Modules and Classes
+------------------------------------------
+
+TorchScript cannot currently compile a number of other commonly used pytorch
+constructs. Below are listed the modules that TorchScript does not support, and
+an incomplete list of pytorch classes that are not supported. For unsupported modules
+we suggest using :meth:`torch.jit.trace`.
+
+  * :class:`torch.nn.RNN`
+  * :class:`torch.nn.AdaptiveLogSoftmaxWithLoss`
```
[the remark this answers is not in the document]
Comment (eellison) [reply docs/source/jit_unsupported.rst]: `CrossMapLRN2d` isn't documented

File: docs/source/jit_unsupported.rst (hunk 1 above)
Comment (eellison) [reply docs/source/jit_unsupported.rst]: Sure, we can do that

File: docs/source/jit_unsupported.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (eellison) [reply docs/source/jit_unsupported.rst]: `CrossMapLRN2d` isn't in our documentation

File: torch/jit/unsupported_tensor_ops.py (hunk 2 above)
Comment (eellison) [reply torch/jit/unsupported_tensor_ops.py]: These are the undocumented apis & properties. I could try to expose the list here: https://github.com/pytorch/pytorch/blob/master/test/test_docs_coverage.py#L83

**Comment by eellison** [comment]
> quip (which itself is missing some other stuff like the unsupported parts of `torch.nn.init`)

What stuff is missing from the quip exactly?

**Comment by fmassa** [comment]
FYI only the non-inplace functions in torch.nn.init are are deprecated

**Pull request closed without merging**